roles/tas_single_node/tasks/certificates.yml

```diff
@@ -136,6 +137,102 @@
         privatekey_path: "{{ tas_single_node_local_ctlog_private_key }}"
         privatekey_passphrase: "{{ tas_single_node_ctlog_ca_passphrase }}"
 
+- name: Handle TSA certificate chain
+  delegate_to: localhost
+  block:
+    - name: Check if TSA root certificate chain exists locally
+      ansible.builtin.stat:
+        path: "{{ tas_single_node_local_tsa_certificate_chain }}"
+      register: tsa_chain_stat
+
+    - name: Check if TSA root private key exists locally
+      ansible.builtin.stat:
+        path: "{{ tas_single_node_local_tsa_private_key }}"
+      when: tas_single_node_signer_type == 'file'
+      register: tsa_private_key_stat
+
+    - name: Create TSA root Private Key
+      community.crypto.openssl_privatekey:
+        path: "{{ tas_single_node_local_tsa_private_key }}"
+        curve: secp256r1
+        type: ECC
+        passphrase: "{{ tas_single_node_tsa_ca_passphrase }}"
+        cipher: auto
+      when: (tsa_private_key_stat.stat.islnk is not defined) and (tsa_chain_stat.stat.islnk is not defined) and tas_single_node_signer_type == 'file'
```

Comment:
I could be wrong here, but I think you can just use `force: false` (see the [module doc](https://docs.ansible.com/ansible/latest/collections/community/crypto/openssl_privatekey_module.html)) to specify that the private key shouldn't be overwritten if it already exists (`false` is also the default, which is why the existing code doesn't seem to address this case at all - it's addressed implicitly).

Reply:
I think it's better to use the same way as the rest of the certificates:
example: 
```
  when: >
    (certs_dir_files.files | selectattr('path', 'equalto', tas_single_node_remote_fulcio_private_key) | list | length) == 0
```
(I updated it accordingly)